src/Functions/FunctionsStringDistance.cpp

```diff
@@ -271,6 +273,180 @@ struct ByteEditDistanceImpl
     }
 };
 
```

Comment:
Curious could we use `iota` to speed up the parts that involve filling up arrays? For example, similar to the recent improvement to [levenshteinDistance](https://github.com/ClickHouse/ClickHouse/blob/master/src/Common/levenshteinDistance.cpp#L15).

Reply:
Actually, https://github.com/ClickHouse/ClickHouse/pull/57288/files#diff-887f9449391c9bd6df6b7d2348fc0f8afcc3f0d9da219d742994c64cc5da91c3 used `iota` in `ByteDamerauLevenshteinDistanceImpl` but I changed to a `for` loop. The code implemented

```
    for i := 0 to length(a) inclusive do
        d[i, 0] := i
    for j := 0 to length(b) inclusive do
        d[0, j] := j
```

from [Wikipedia](https://en.wikipedia.org/wiki/Damerau%E2%80%93Levenshtein_distance#Optimal_string_alignment_distance) while iota might be a tiny bit faster (perhaps it isn't because the compiler optimizes simple `for` loops as well, e.g. unrolling/vectorization) a `for` loop is more consistent with Wikipedia and within the code internally.

That is the only place eligible for `iota` as far as I see.